internal/run/run.go

```diff
@@ -826,6 +829,105 @@ func (w *Workflow) snapshotSource(ctx context.Context, parentStep *workflowTrack
 	return nil
 }
 
+func (w *Workflow) snapshotCodeSamples(ctx context.Context, parentStep *workflowTracking.WorkflowStep, overlayString string, codeSampleConfig workflow.CodeSamples) error {
+	registryLocation := codeSampleConfig.Registry
+
+	if registryLocation == nil {
+		return nil
+	}
+
+	tags := []string{"latest"}
+	if env.IsGithubAction() {
```

Comment:
should we factor this out. Since I have similar code in that registyTags function